samples/azure-quantum/hidden-shift-gallery/hidden-shift.ipynb

```diff
@@ -0,0 +1,1085 @@
+        "exponential advantage over the best known classical algorithms. \n",
+        "This may eventually help in solving deconvolution problems and enable us to efficiently find patterns in complex data sets,\n",
+        "for example finding the pattern on the right-hand side in the graph:\n",
+        "\n",
+        "<img src=\"hw-shift-figure.png\" />\n",
+        "\n",
+        "You can have a look into [arXiv:0812.0380](https://arxiv.org/abs/0812.0380) for general background on hidden shifts and related problems\n",
+        "and into [arXiv:0811.3208](https://arxiv.org/abs/0811.3208) for the case of hidden shifts over Boolean functions.\n",
+        "\n",
+        "We will investigate the latter case, in which one is given two Boolean bent functions $f$ and $g$ with the promise that they satisfy the relation\n",
+        "\n",
+        "$$ g(x) = f(x \\oplus s) $$\n",
+        "\n",
+        "for all $x$, where $s$ is a hidden bitstring, called _shift_ that we would like to find.\n",
+        "A Boolean function is _bent_ if it is as far from linear as possible. In\n",
+        "particular, bent functions have flat Fourier spectra, such that each\n",
+        "Fourier coefficient is equal in absolute value.\n",
+        "\n",
+        "In this case, the quantum algorithm described in [arXiv:0811.3208](https://arxiv.org/abs/0811.3208) uses\n",
+        "black-box phase oracles $U_{f^\\star}$ and $U_g$ for $f^\\star$ and $g$, respectively, where $f^\\star$ is the dual bent function\n",
+        "to $f$ (defined in more detail below), and computes the hidden shift $s$ between $f$ and $g$.\n",
+        "\n",
+        "The quantum circuit for this type of hidden shift problems is as follows:\n",
+        "\n",
+        "<img src=\"hidden-shift-hs1.png\" width=\"600px\" />\n",
+        "\n",
+        "Black-box phase oracles compute the value of a Boolean function $f(x)$ into the phase, i.e., $U_f = \\sum_x (-1)^{f(x)}|x\\rangle\\langle x|$.\n",
+        "More information on phase oracles can be found in the [Quantum Katas](https://github.com/microsoft/QuantumKatas/tree/main/tutorials/Oracles).\n",
+        "\n",
+        "You can have a look into [arXiv:0812.0380](https://arxiv.org/abs/0812.0380) for general background on hidden shifts and related problems\n",
+        "and into [arXiv:0811.3208](https://arxiv.org/abs/0811.3208) for the case of hidden shifts\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## 1. Connect to the Azure Quantum workspace\n",
+        "\n",
+        "Let's start by connecting to the Azure Quantum service.\n",
+        "We construct an instance of the `AzureQuantumProvider`. Note that it's imported from `azure.quantum.qiskit`."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "dotnet_interactive": {
```

Comment:
VS Code. Do you know an easy way to remove it?

Reply:
Ah, that makes sense. May not need to remove it, then, since that pretty closely matches the experience in-portal. If in the future you do need to remove this metadata, could possibly do so by rerunning in Jupyter Notebook or Lab directly.